src/ts-ast.ts

```diff
@@ -298,25 +298,47 @@ export abstract class FunctionLike {
   description: string;
   params: Param[];
   returns: Type;
+  returnsDescription: string;
 
   constructor(data: {
     name: string,
     description?: string,
     params?: Param[],
-    returns?: Type
+    returns?: Type,
+    returnsDescription?: string
   }) {
     this.name = data.name;
     this.description = data.description || '';
     this.params = data.params || [];
     this.returns = data.returns || anyType;
+    this.returnsDescription = data.returnsDescription || '';
   }
 
   serialize(depth: number = 0): string {
     let out = ''
     const i = indent(depth);
-    if (this.description) {
-      out += '\n' + formatComment(this.description, depth);
+
+    const annotations = [];
+    for (const p of this.params) {
+      if (p.description) {
+        annotations.push(`@param ${p.name} ${p.description}`);
+      }
+    }
+    if (this.returnsDescription) {
+      annotations.push(`@returns ${this.returnsDescription}`);
+    }
+
+    let combinedDescription = this.description;
+    if (annotations.length > 0) {
+      if (combinedDescription) {
+        combinedDescription += '\n\n';
+      }
+      combinedDescription += annotations.join('\n');
+    }
+    if (combinedDescription) {
```

Comment:
I'm preferring more explicit conditions these days, check length?

Reply:
Open to it. I use boolean coercion for testing empty strings all throughout this project, so would like to decide if that's the style we should starting doing everywhere.